Makefile

```diff
@@ -202,6 +202,22 @@ THIRD_PARTY_NOTICE_LICENSE_TYPE = microsoft-oss
 $(eval $(call CREATE_THIRD_PARTY_NOTICES_RULE,ThirdPartyNotices.txt,foundation,False,False))
 $(eval $(call CREATE_THIRD_PARTY_NOTICES_RULE,bin/$(CONFIGURATION)/lib/xamarin.android/ThirdPartyNotices.txt,$(THIRD_PARTY_NOTICE_LICENSE_TYPE),True,False))
 
+EXTERNAL_GIT_PATH=$(topdir)/external
+XA_PATH=$(topdir)
+prepare-external-git-dependencies:
+	$(call MSBUILD_BINLOG,prep-tasks) build-tools/xa-prep-tasks/xa-prep-tasks.csproj
+	$(call MSBUILD_BINLOG,external-checkout) build-tools/xa-prep-tasks/xa-prep-tasks.csproj \
+		/t:CheckoutExternalGitSources /p:ExternalSourceDependencyDirectory='$(EXTERNAL_GIT_PATH)'
+
+prepare-commercial: prepare-external-git-dependencies
+	cd $(EXTERNAL_GIT_PATH)/monodroid && ./configure --with-xamarin-android='$(XA_PATH)'
+
+commercial:
+	make -C $(EXTERNAL_GIT_PATH)/monodroid commercial-xa XA_TOP_SOURCE_DIRECTORY='$(XA_PATH)' CONFIGURATION=$(CONFIGURATION)
+
+commercial-installers:
```

Comment:
I'm not sure I understand the major need for the `commercial-installers` target, or `prepare-commercial`, `commercial`.

Thought experiment: could we instead use conditional [makefile inclusion](https://www.gnu.org/software/make/manual/html_node/Include.html)?

```make
-include $(EXTERNAL_GIT_PATH)/monodroid/xa-integration.mk
```

`xa-integration.mk` could then "hook" into `prepare::`, `all::`, `all-tests::`, etc. targets.

Regarding `commercial-installers` in particular, it appears the *primary* aspect is to allow overriding the location of the `License.txt` to include in the installers.  If we go with a `-include file.mk` approach, we could use a Make variable to provide a default and override that as appropriate, e.g.

```make
#  Makefile
LICENSE = OSS.txt

-include Proprietary.mk

all::
	echo LICENSE=$(LICENSE)
```

```make
# Proprietary.mk
LICENSE = Proprietary.txt
```

Running `make -f Makefile` will result in `LICENSE=Proprietary.txt` being printed.

Reply:
Great idea, I like this approach. Corresponding updates have been made.